File: Games/GameOfGames/src/main/java/jokatu/game/games/gameofgames/game/GameOfGames.java
package jokatu.game.games.gameofgames.game;

import jokatu.game.Game;
import jokatu.game.GameID;
import jokatu.game.player.StandardPlayer;
import jokatu.game.stage.machine.SingleStageMachine;
import org.jetbrains.annotations.NotNull;

/**
 * The game of games.  The beginning.  The end.
 *
 * A game you play to create games.
 */
public class GameOfGames extends Game<StandardPlayer> {

	public static final String GAME_OF_GAMES = "Game of games";

	GameOfGames(GameID gameID) {
		super(gameID);

		stageMachine = new SingleStageMachine(new GameOfGameStage());
	}

	@NotNull
	@Override
	public String getGameName() {
		return GAME_OF_GAMES;
	}
}

File: Games/GameOfGames/src/main/kotlin/jokatu/game/games/gameofgames/game/GameOfGames.kt
package jokatu.game.games.gameofgames.game

import jokatu.game.Game
import jokatu.game.GameID
import jokatu.game.player.StandardPlayer
import jokatu.game.stage.machine.SingleStageMachine

/**
 * The game of games.  The beginning.  The end.
 *
 * A game you play to create games.
 */
class GameOfGames internal constructor(gameID: GameID) : Game<StandardPlayer>(gameID) {

	override val gameName: String
		get() = GAME_OF_GAMES

	init {
		stageMachine = SingleStageMachine(GameOfGameStage())
	}

	companion object {
		const val GAME_OF_GAMES = "Game of games"
	}
}
